tests/posterior_parameters_test.py

```diff
@@ -385,3 +385,12 @@ def test_invalid_literal_field_values():
     """
 
     MCMCPosteriorParameters(method="invalid")
+
+
+def test_if_warning_raised_for_deprecated_build_posterior_parameters(get_inference):
+    """
+    Check if the build_posterior method raises a warning for deprecated parameters
+    """
+
+    with pytest.warns(DeprecationWarning):
+        get_inference.build_posterior(mcmc_parameters={})
```

Comment:
add a case with non-default values as well please, e.g., `thin=10`, and check how the warning is displayed, whether everything makes sense.